lib/intacct_ruby/response.rb

```diff
@@ -13,24 +13,38 @@ module IntacctRuby
   class Response
     attr_reader :response_body
 
-    def initialize(xml_response)
-      @response_body = Nokogiri::XML(xml_response.body)
+    def initialize(http_response, json: false)
+      @json = json
+
+      if json?
+        @response_body = JSON.parse(http_response.body)
```

Comment:
Can we determine a JSON response based on the content type of `http_response`? Looks like IntacctRuby is using `Net::Http` under the hood https://github.com/privateprep/intacct-ruby/blob/5469ecf5246ccd628671276bf1592f3cd9f55293/lib/intacct_ruby/api.rb#L26